src/utils.ts

```diff
@@ -0,0 +1,275 @@
+      return encodeURIComponent(key) + '=' +
+        encodeURIComponent(json[key]);
+    }).join('&');
+}
+
+
+export
+interface IAjaxSuccess {
+  data: any;
+  status: string;
+  xhr: XMLHttpRequest;
+};
+
+
+export
+interface IAjaxError {
+  xhr: XMLHttpRequest;
+  status: string;
+  err: string;
+};
+
+
+export
+interface IAjaxSetttings {
+  method: string;
+  dataType: string;
+  contentType?: string;
+  data?: any;
+};
+
+/*
+ * Asynchronous XMLHTTPRequest handler
+ *
+ * http://www.html5rocks.com/en/tutorials/es6/promises/#toc-promisifying-xmlhttprequest
+ */
+export
+var ajaxProxy = function(url: string, settings: IAjaxSetttings): Promise<any> {
+
+  return new Promise(function(resolve, reject) {
+    var req = new XMLHttpRequest();
+    req.open(settings.method, url);
+    if (settings.contentType) {
+      req.overrideMimeType(settings.contentType);
+    }
+
+    req.onload = function() {
+      if (req.status == 200) {
+        if (settings.dataType === 'json') {
+          resolve(JSON.parse(req.response));
+        }
+        else {
+          resolve(req.response);
+        }
+      }
+      else {
+        reject(req.statusText);
+      }
+    }
+
+    req.onerror = function() {
+      reject(req.statusText);
+    }
+
+    if (settings.data) {
+      req.send(settings.data);
+    } else {
+      req.send();
+    }
+  });
+};
+
+/**
+ * log ajax failures with informative messages
+ */
+export
+var logAjaxError = function(status: string) {
+  var msg = "API request failed (" + status + "): ";
+  console.log(msg);
+}
+
+
+declare function require(modules: string[], success: Function, reject: Function): void;
+
+/**
+ * Tries to load a class
+ *
+ * Tries to load a class from a module using require.js, if a module 
+ * is specified, otherwise tries to load a class from the global 
+ * registry, if the global registry is provided.
+ */
+export
+var loadClass = function(class_name: string, module_name: string, registry: { [string: string]: Function; }) {
+  return new Promise(function(resolve, reject) {
+
+    // Try loading the view module using require.js
+    if (module_name) {
+      require([module_name], function(module: any) {
+        if (module[class_name] === undefined) {
+          reject(new Error('Class ' + class_name + ' not found in module ' + module_name));
+        } else {
+          resolve(module[class_name]);
+        }
+      }, reject);
+    } else {
+      if (registry && registry[class_name]) {
+        resolve(registry[class_name]);
+      } else {
+        reject(new Error('Class ' + class_name + ' not found in registry '));
+      }
+    }
+  });
+};
+
+
+/**
+ * Creates a wrappable Promise rejection function.
+ * 
+ * Creates a function that returns a Promise.reject with a new WrappedError
+ * that has the provided message and wraps the original error that 
+ * caused the promise to reject.
+ */
+export
+  var reject = function(message: string, log?: boolean): (error: Error) => Promise<any> {
```

Comment:
Can you point me to where this function is used? I don't fully understand its purpose.